Add a section on requirements for building the repo on Alpine Linux (#9924)
microsoft/aspire: docs/machine-requirements.md

This will take on the free version of Codespace around 10 mins.

> :warning: With the free version of Codespaces the development experience can be less than ideal. We recommend using at least a Codespace with 16GB of RAM or use your local VS Code / DevContainers instance.

## Alpine Linux

If you want to build the Aspire repo on Alpine Linux, you'll need to build or install musl compatible gRPC tooling as the `Grpc.Tools` package Aspires uses to generate gRPC interfaces depends on native binaries, but doesn't include a musl specific build.

On Alpine Linux, the `grpc-plugins` package includes the necessary binaries. You can install it with:

```bash
apk add --no-cache grpc-plugins
```

> You'll need to override the default `Grpc.Tools` binary paths to point to your musl compatible gRPC binaries using the following environment variables (assuming you installed the `grpc-plugins` package):

Reviewer: ```suggestion
Then, override the default `Grpc.Tools` binary paths to point to your musl-compatible gRPC binaries by setting the following environment variables:
```

```bash
export PROTOBUF_PROTOC=/usr/bin/protoc
export GRPC_PROTOC_PLUGIN=/usr/bin/grpc_csharp_plugin
```

With that, you can build and run the Aspire repo on Alpine Linux.

> :warning: Aspire currently only directly supports the x64/amd64 architecture for Alpine/musl. If you want to build or run Aspire in Alpine on arm64, you may need to use an arm64/x64 compatibility layer like `qemu`.
>

> > :warning: Alpine Linux support was added in commit https://github.com/dotnet/aspire/commit/cc2706a90848deec90aa166054e1b2a4ecf94689 and isn't supported in earlier releases. Additionally, Alpine Linux is not currently part of our CI test suite.

Reviewer: ```suggestion
> :warning: Alpine Linux support was added in [this commit](https://github.com/dotnet/aspire/commit/cc2706a90848deec90aa166054e1b2a4ecf94689) and isn't supported in earlier releases. Additionally, Alpine Linux is not currently part of our CI test suite.
```
